pkg/event-manager/drivers/docker_backend.go

```diff
@@ -95,32 +101,49 @@ func (d *dockerBackend) Deploy(ctx context.Context, driver *entities.Driver) err
 	io.Copy(os.Stdout, rc)
 
 	return utils.Backoff(time.Duration(d.DeployTimeout)*time.Second, func() error {
-		created, err := d.dockerClient.ContainerCreate(ctx, &container.Config{
+		config := &container.Config{
 			Image: driver.Image,
 			Env:   bindEnv(secrets),
 			Cmd:   buildArgs(driver.Config),
 			Labels: map[string]string{
 				labelEventDriverID: driver.ID,
 			},
-		}, nil, nil, "")
+		}
+
+		freePort, err := GetFreePort()
+		if err != nil {
+			return errors.Wrap(err, "error get free port when creating container")
+		}
+
+		hostConfig := &container.HostConfig{}
+
+		if driver.Expose {
+			config.ExposedPorts = nat.PortSet{
+				"80/tcp": struct{}{},
+			}
+			hostConfig.PortBindings = nat.PortMap{
+				"80/tcp": []nat.PortBinding{
+					{
+						HostIP:   "0.0.0.0",
+						HostPort: strconv.Itoa(freePort),
+					},
+				},
+			}
+		}
+
+		created, err := d.dockerClient.ContainerCreate(ctx, config, hostConfig, nil, "")
 		if err != nil {
 			return errors.Wrap(err, "error creating container")
 		}
 
 		if err := d.dockerClient.ContainerStart(ctx, created.ID, types.ContainerStartOptions{}); err != nil {
 			return errors.Wrap(err, "error starting container")
 		}
+		driver.URL = fmt.Sprintf("http://127.0.0.1:%d", freePort)
```

Comment:
Where will driver.URL be referred later? Driver and solo will be running on same host, may i know why the URL is always 12.0.0.1 ?

Reply:
Just for print, when do `dispatch get event`, I think it should be only return the port, then dispatch cli will read the config and output full url. How do you think about this?